### PR TITLE
Impr docs on batch sizes and limits in distributed

### LINKED ISSUE #21022: Trainer parameter limit-train-batches was meant to be per-worker
### 📚 Documentation

When using `lightning` with `RayDDPStrategy`, we found that `Trainer`'s `limit_train_batches` parameter was actually meant for each worker instead of "global".

I wanna confirm with the developer if
1. this is the case with other parallel training strategies
2. this applies to other `limit_*_batches` parameters of the `Trainer`.

It will be great to update the doc to mention this, e.g., 
https://lightning.ai/docs/pytorch/stable/common/trainer.html#limit-train-batches

Thanks

cc @lantiga @borda @justusschock

### LINKED ISSUE #21023: DDP Strategy Does Not Automatically Shard Batch Sizes Despite Documentation Claims
### Bug description

# DDP Strategy Does Not Automatically Shard Batch Sizes Despite Documentation Claims

## Issue Description

The Lightning documentation claims that DDP automatically shards both datasets and batch sizes, but experimental testing shows that only dataset sharding works automatically. Batch sizes are not automatically divided across GPUs, preventing users from achieving expected speedup from multi-GPU training.

## Expected Behavior (According to Documentation)

According to [the GPU training documentation](https://lightning.ai/docs/pytorch/stable/accelerators/gpu_intermediate.html):

> "Each GPU gets visibility into a subset of the overall dataset. It will only ever see that subset."

This suggests that DDP should automatically:
1. Shard the dataset (different data per GPU) ✅ **Works**
2. Shard the batch size (smaller batches per GPU) ❌ **Does NOT work**

## Actual Behavior

When using `strategy="ddp"` or `DDPStrategy()` with multiple GPUs:

- **Dataset sharding works**: Each GPU gets different data (confirmed by different input/target values)
- **Gradient synchronization works**: Weight matrices remain identical across GPUs (confirmed by weight norm checks)
- **Batch size sharding does NOT work**: Each GPU processes the full batch size instead of `batch_size / num_gpus`

## Reproduction Steps

```python
import lightning.pytorch as pl
from lightning.pytorch.strategies import DDPStrategy
from pytorch_lightning import LightningModule, Trainer
from torch import nn
from torch.utils.data import DataLoader, TensorDataset
import torch

class SimpleModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = nn.Linear(10, 1)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = nn.functional.mse_loss(y_hat, y)
        
        if batch_idx == 0:
            print(f"[Rank {self.global_rank}] Batch shape: {x.shape}")
            print(f"[Rank {self.global_rank}] First element: {x[0, 0].item():.6f}")
        
        return loss

    def configure_optimizers(self):
        return torch.optim.SGD(self.parameters(), lr=0.01)

# Create dataset and dataloader
X = torch.randn(1000, 10)
y = torch.randn(1000, 1)
ds = TensorDataset(X, y)
dl = DataLoader(ds, batch_size=64, shuffle=False, num_workers=0)

model = SimpleModel()
strategy = DDPStrategy(find_unused_parameters=False, static_graph=True)

trainer = Trainer(
    accelerator="gpu",
    devices=2,
    strategy=strategy,
    max_epochs=1,
    logger=False,
    enable_checkpointing=False,
)

trainer.fit(model, dl)
```

## Expected vs Actual Output

**Expected (if batch sharding worked):**
```
[Rank 0] Batch shape: torch.Size([32, 10])  # 64/2 = 32
[Rank 1] Batch shape: torch.Size([32, 10])  # 64/2 = 32
```

**Actual output:**
```
[Rank 0] Batch shape: torch.Size([64, 10])  # Full batch size
[Rank 1] Batch shape: torch.Size([64, 10])  # Full batch size
```

## Impact

1. **No speedup from multi-GPU training**: Each GPU uses full memory instead of reduced memory
2. **Memory pressure**: Users hit OOM errors that could be avoided with proper batch sharding
3. **Misleading documentation**: Users spend hours debugging why they're not getting expected performance
4. **Manual workaround required**: Users must manually divide batch sizes by number of GPUs

## Workaround

Users must manually reduce batch sizes in DataLoaders:

```python
# For 2 GPUs, use batch_size=32 instead of 64
dl = DataLoader(ds, batch_size=32, shuffle=False, num_workers=0)
# Effective batch size = 32 * 2 = 64
```

## Request

Either:
1. **Fix automatic batch sharding** to match documentation claims, OR
2. **Update documentation** to clarify that only dataset sharding is automatic and batch sharding requires manual configuration

## Environment

- PyTorch Lightning version: [latest]
- PyTorch version: [latest]
- CUDA version: [latest]
- Number of GPUs: 2

This issue cost several hours of debugging time and should be addressed to prevent other users from experiencing the same frustration.

### What version are you seeing the problem on?

v2.5

### Reproduced in studio

_No response_

### How to reproduce the bug

```python
import lightning.pytorch as pl
from lightning.pytorch.strategies import DDPStrategy
from pytorch_lightning import LightningModule, Trainer
from torch import nn
from torch.utils.data import DataLoader, TensorDataset
import torch

class SimpleModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = nn.Linear(10, 1)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = nn.functional.mse_loss(y_hat, y)
        
        if batch_idx == 0:
            print(f"[Rank {self.global_rank}] Batch shape: {x.shape}")
            print(f"[Rank {self.global_rank}] First element: {x[0, 0].item():.6f}")
        
        return loss

    def configure_optimizers(self):
        return torch.optim.SGD(self.parameters(), lr=0.01)

# Create dataset and dataloader
X = torch.randn(1000, 10)
y = torch.randn(1000, 1)
ds = TensorDataset(X, y)
dl = DataLoader(ds, batch_size=64, shuffle=False, num_workers=0)

model = SimpleModel()
strategy = DDPStrategy(find_unused_parameters=False, static_graph=True)

trainer = Trainer(
    accelerator="gpu",
    devices=2,
    strategy=strategy,
    max_epochs=1,
    logger=False,
    enable_checkpointing=False,
)

trainer.fit(model, dl)
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version (e.g., 2.5.0):
#- PyTorch Version (e.g., 2.5):
#- Python version (e.g., 3.12):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source): pip
```
</details>
Here is my slurm submission: 
```bash
#!/bin/bash
#SBATCH --job-name=mwe_2gpu
#SBATCH --output=mwe-%x-%j.out
#SBATCH --error=mwe-%x-%j.err
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=2
#SBATCH --gres=gpu:2
#SBATCH --mem=60GB
#SBATCH --time=01:00:00
#SBATCH -A pilotgpu


# Ensure proper SLURM environment variables for PyTorch Lightning
export SLURM_NTASKS_PER_NODE=2

# Print cluster info for debugging
echo "=== SLURM Job Info ==="
echo "Job ID: $SLURM_JOB_ID"
echo "Nodes: $SLURM_JOB_NUM_NODES"
echo "Tasks per node: $SLURM_NTASKS_PER_NODE"
echo "CPUs per task: $SLURM_CPUS_PER_TASK"
echo "GPUs requested: $SLURM_GPUS_PER_NODE"
echo "Available GPUs: $CUDA_VISIBLE_DEVICES"
echo "======================="

# Load modules and setup environment
module load cuda/12.1 || echo "Warning: Could not load CUDA module"
module load webproxy
pip install torch lightning

# nproc_per_node == number of GPUs (or else won't work)
torchrun \
    --nproc_per_node=2 \
    --nnodes=1 \
    --node_rank=0 \
    --master_addr=localhost \
    --master_port=12355 \
    mwe.py
```

### More info

I would like confirmation about the batch size under DDS. If the batch size is 64 on a single GPU, and I run in DDS mode, should the batch size  e 32 on each GPU, or 64 on each GPU?

cc @justusschock @lantiga

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #21023
Fixes #21022

The two issues linked was raised back-to-back basically contains the same confusion:

*What happens in distributed settings with the batch size?*

One is related to the effective batch size in distributed settings the other to `limit_*_batches` arguments. In both cases I have tried to make it clear that in distributed settings the values are not global but instead per device.

Connected to this when i tried to look for the documentation on hardware acceleration I could not initially find it. It was due to me not connecting the headline `Interactive cloud development` to hardware acceleration. I guess it was originally named that to highlight lightning.ai as a service, but that is not mention it the docs. So the headline feels very confusing. If this is too much, I can revert this change.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21070.org.readthedocs.build/en/21070/

### CODE REVIEW

File: docs/source-pytorch/accelerators/gpu_faq.rst [hunk 1]
```diff
@@ -5,31 +5,71 @@
...
-.. testcode::
+    \text{Effective Batch Size} = \text{DataLoader Batch Size} \times \text{Number of Devices} \times \text{Number of Nodes}
 
-    class LitModel(LightningModule):
-        def train_dataloader(self):
-            return Dataset(..., batch_size=7)
+A couple of examples to illustrate this:
 
-Whenever you use multiple devices and/or nodes, your effective batch size will be 7 * devices * num_nodes.
+.. testcode::
```
Comment (Borda) [inline docs/source-pytorch/accelerators/gpu_faq.rst]:
```suggestion
.. code-block:: python
```

File: docs/source-pytorch/accelerators/gpu_faq.rst [hunk 2]
```diff
@@ -5,31 +5,71 @@
...
+
+Because the effective batch size is larger when using multiple devices, you need to adjust your learning rate
+accordingly. Because the learning rate is a hyperparameter that controls how much to change the model in response to
+the estimated error each time the model weights are updated, it is important to scale it with the effective batch size.
+
+In general, there are two common scaling rules:
+
+1. **Linear scaling**: Increase the learning rate linearly with the number of devices.
+
+    .. testcode::
```
Comment (Borda) [inline docs/source-pytorch/accelerators/gpu_faq.rst]:
```suggestion
    .. code-block:: python
```

File: docs/source-pytorch/accelerators/gpu_faq.rst [hunk 3]
```diff
@@ -5,31 +5,71 @@
...
+
+        # Example: Linear scaling
+        base_lr = 1e-3
+        num_devices = 8
+        scaled_lr = base_lr * num_devices  # 8e-3
+
+2. **Square root scaling**: Increase the learning rate by the square root of the number of devices.
+
+    .. testcode::
```
Comment (Borda) [inline docs/source-pytorch/accelerators/gpu_faq.rst]:
```suggestion
    .. code-block:: python
```

**Review by lantiga — APPROVED** [review]
Looks good!
